client/state/data-layer/wpcom/user-settings/index.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * External dependencies
+ */
+import { isEmpty, keys, merge, noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { decodeEntities } from 'lib/formatting';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { updateUserSettings, clearUnsavedUserSettings } from 'state/user-settings/actions';
+import {
+	USER_SETTINGS_REQUEST,
+	USER_SETTINGS_SAVE,
+} from 'state/action-types';
+
+/*
+ * Decodes entities in those specific user settings properties
+ * that the REST API returns already HTML-encoded
+ */
+function fromApi( apiResponse ) {
+	const decodedValues = {
+		display_name: apiResponse.display_name && decodeEntities( apiResponse.display_name ),
+		description: apiResponse.description && decodeEntities( apiResponse.description ),
+		user_URL: apiResponse.user_URL && decodeEntities( apiResponse.user_URL )
+	};
+
+	return merge( {}, apiResponse, decodedValues );
+}
+
+export const requestUserSettings = ( { dispatch }, action, next ) => {
+	dispatch( http( {
+		apiVersion: '1.1',
+		method: 'GET',
+		path: '/me/settings',
+	}, action ) );
+
+	return next( action );
+};
+
+export const receiveUserSettings = ( { dispatch }, action, next, data ) => {
+	dispatch( updateUserSettings( fromApi( data ) ) );
+};
+
+export const receiveRequestError = noop;
+
+/*
+ * Post settings to WordPress.com API at /me/settings endpoint
+ */
+export function saveUserSettings( { dispatch, getState }, action, next ) {
+	const { settingsOverride } = action;
+	const settings = settingsOverride || getState().userSettings.unsavedSettings;
+
+	if ( ! isEmpty( settings ) ) {
+		dispatch( http( {
+			apiVersion: '1.1',
+			method: 'POST',
+			path: '/me/settings',
+			body: settings,
+		}, action ) );
+	}
+
+	return next( action );
+}
+
+export const receiveSavedUserSettings = ( { dispatch }, { settingsOverride }, next, data ) => {
```

Comment:
I believe that the purpose of the `receive` handlers really is to _receive_ things. They are invoked when a response from the server arrives, and their `data` parameter is in the wire format. They are clearly a part of the data layer.

The `receive` handler further dispatches one or more actions, and it's these actions that do the actual _updating_ or _storing_, not the handler itself.

If I called the `receiveSavedUserSettings` in the devtools REPL, the meaning would be: here is a JSON POST response we received from server, please do the work that needs to be done: display notices, update the store data, clear unsaved state on a form, redirect to a different page...

If I wanted to do just the store update from the REPL, I would directly dispatch the `updateUserSettings` action.